docs/devcontainerjson-reference.md

```diff
@@ -0,0 +1,172 @@
+# devcontainer.json reference
+
+A `devcontainer.json` file in your project tells `devcontainer.json` supporting services / tools how to access (or create) a **development container** with a well-defined tool and runtime stack. It's currently supported by the [Remote - Containers](https://aka.ms/vscode-remote/download/containers) extension and [GitHub Codespaces](https://github.com/features/codespaces).
+
+[Set up a folder to run in a container](https://code.visualstudio.com/docs/remote/create-dev-container#_set-up-a-folder-to-run-in-a-container) has more information on configuring a dev container or you can use the **Remote-Containers: Add Development Container Configuration Files...** or **Codespaces: Add Development Container Configuration Files...** commands from the Command Palette (`kbstyle(F1)`) to add a wide variety of base configurations from the [vscode-dev-containers repository](https://github.com/microsoft/vscode-dev-containers/tree/main/containers).
+
+> **Tip:** If you've already built a container and connected to it, be sure to run **Remote-Containers: Rebuild Container** or **Codespaces: Rebuild Container** from the Command Palette (`kbstyle(F1)`) to pick up any changes you make.
+
+## Scenario specific properties
+
+The focus of `devcontainer.json` is to describe how to enrich a container for the purposes of development rather than acting as a multi-container orchestrator format. Instead, container orchestrator formats can be referenced when needed to manage multiple containers and their lifecycles. Today, `devcontainer.json` includes scenario specific properties for working without a container orchestrator (by directly referencing an image or Dockerfile) and for using Docker Compose as a simple multi-container orchestrator.
+
+### Image or Dockerfile specific properties
+
+| Property | Type | Description |
+|----------|------|-------------|
+| `image` | string | **Required** when [using an image](https://code.visualstudio.com/docs/remote/create-dev-container#_using-an-image-or-dockerfile). The name of an image in a container registry ([DockerHub](https://hub.docker.com), [GitHub Container Registry](https://docs.github.com/packages/guides/about-github-container-registry), [Azure Container Registry](https://azure.microsoft.com/services/container-registry/)) that `devcontainer.json` supporting services / tools should use to create the dev container. |
+| `build.dockerfile` / `dockerFile`  | string |**Required** when [using a Dockerfile](https://code.visualstudio.com/docs/remote/create-dev-container#_dockerfile). The location of a [Dockerfile](https://docs.docker.com/engine/reference/builder/) that defines the contents of the container. The path is relative to the `devcontainer.json` file. You can find Dockerfiles for different runtimes in the [vscode-dev-containers repository](https://github.com/microsoft/vscode-dev-containers/tree/main/containers). |
```

Comment:
Let's drop the old `dockerFile` property from here along with the `context` property on the next line and just keep the `build.XXX`.  I'd also suggest removing the reference to the vscode-dev-containers repository.

Reply:
Updated following guidance above! 

The `build.dockerFile` property linked to [VS Code docs](https://code.visualstudio.com/docs/remote/create-dev-container#_dockerfile), so also removed that.